spec/controllers/ops_controller_spec.rb

```diff
@@ -213,7 +213,7 @@
                                   :name        => "not the default",
                                   :description => "Not the Default Zone")
 
-        current = double("current", :[] => {:server => {:zone => "default"}}).as_null_object
+        current = Settings.to_hash.deep_merge(:server => {:zone => "default"})
```

Comment:
This change is weird, but it was failing on Ruby 2.7 (not 3.0) with the following:

```
  2) OpsController#settings_update when the zone is changed updates the server's zone
     Failure/Error: current = current.deep_clone

     RuntimeError:
       #<Class:#<RSpec::Mocks::Double:0x0000562147dfb178>>#marshal_dump returned same class instance
     # ./app/controllers/application_controller.rb:1448:in `build_audit_payload_changes'
     # ./app/controllers/application_controller.rb:1431:in `build_audit_payload'
     # ./app/controllers/application_controller.rb:1423:in `build_config_audit'
     # ./app/controllers/ops_controller/settings/common.rb:424:in `settings_update_save'
     # ./spec/controllers/ops_controller_spec.rb:231:in `block (5 levels) in <top (required)>'
     # ./spec/controllers/ops_controller_spec.rb:231:in `block (4 levels) in <top (required)>'
```

In the 2.7 tests, RSpec's double is not deep-clonable, but I guess it is in 3.0?  Either way, the "current" should be the full contents, test-wise, so this change is better regardless.

Reply:
Interesting.  I know that expectations using `with` had issues with 2.7/3.0 compatibility where you needed to be explicit if it's a hash or kwargs, I wonder if that's somewhat related.